assignments/Quokka/attemps.js

```diff
@@ -0,0 +1,445 @@
+		"car_model": "Magnum",
+		"car_year": 2008
+	},
+	{
+		"id": 49,
+		"car_make": "Chrysler",
+		"car_model": "Sebring",
+		"car_year": 1996
+	},
+	{
+		"id": 50,
+		"car_make": "Lincoln",
+		"car_model": "Town Car",
+		"car_year": 1999
+	}
+];
+
+
+// Example for loop:
+
+// arr = [1,2,3,4];
+// for (let i = 0; i < arr.length; i++) {
+//     arr[i]; // 1,2,3,4
+// }
+
+// ==== Challenge 1 ====
+// The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
+
+//My Attempt
+console.log(`Car 33 is a ${inventory[32].car_year} ${inventory[32].car_make} ${inventory[32].car_model}`);
+
+// Jonathan's Answer
+// for(const car of inventory){
+// 	if(car.id === 33){
+// 		console.log(`Car 33 is a ${car.car_year} ${car.car_make} ${car.car_model}`);
+// 	}
+// }
+
+
+// ==== Challenge 2 ====
+// The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
+
+// My Attempt
+let lastCar = `${inventory[inventory.length - 1].car_make} ${inventory[inventory.length - 1].car_model}`;
+console.log(lastCar);
+
+//Jonathans Answer
+let lastCar = inventory[inventory.length - 1];
+console.log(`Car ${lastCar.id} is a ${lastCar.car_make} ${lastCar.car_model}`);
+
+// ==== Challenge 3 ====
+// The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
+let carModels = [];
+
+//My Attempt
+for (i = 0; i < inventory.length; i++) {
+	carModels.push(inventory[i].car_model);
+}
+console.log(carModels.sort());
+
+
+// Jonathans Answer
+//Imperative
+// for(const car of inventory) carModels.push(car.car_model);
+// carModels.sort();
+
+// console.log(carModels);
+
+//Declaritive
+carModels = inventory.map(car => car.car_model).sort();
+
+console.log(carModels);
+
+// ==== Challenge 4 ====
+// The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
+let carYears = [];
+
+
+//My Attempt
+
+for (i = 0; i < inventory.length; i++) {
+	carYears.push(inventory[i].car_year);
+}
+console.log(carYears.sort());
+
+//Jonathans Attempt
+
+//Imperative
+for(const car of inventory) carYears.push(car.car_year);
+console.log(carYears);
+
+
+//Declaratvie
+
+carYears = inventory.map(car => car.car_year);
+console.log(carYears);
+
+// ==== Challenge 5 ====
+// The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
+let oldCars = [];
+
+//My attempt
+carYears.filter((i) => {
+	if (i < 2000) {
+		oldCars.push(i)
+	};
+});
```

Comment:
A `.forEach` method would be better suited for what you're attempting to do here!

Reply:
I'll look into it. I really liked you for(const car in inventory) loop, I want to practice that more.